server/api/util.go

```diff
@@ -127,17 +127,18 @@ func patchJSON(client *http.Client, url string, body []byte) error {
 	return nil
 }
 
-func doDelete(client *http.Client, url string) (*http.Response, error) {
-	req, err := http.NewRequest("DELETE", url, nil)
+func doDelete(client *http.Client, url string) (int, error) {
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
 	if err != nil {
-		return nil, err
+		return http.StatusBadRequest, err
 	}
+	// TODO: find a better way to return status code.
 	res, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return http.StatusInternalServerError, err
```

Comment:
can we `return 0, err` here?

Reply:
The caller will get 0 as the status code which is not good. I think an alternative is to return the status code according to the detailed error. But I'm going to leave it for now.

Reply:
0 is ok to use as an _invalid status code_ when err is not nil. If you construct status code in client side, we cannot distiguish the cases that we receive the status code from server or construct it from client.

Reply:
Make sense, change to 0.